tests/user.handles.test.ts

```diff
@@ -0,0 +1,169 @@
+
+beforeAll(async () => {
+  await DatabaseConnection.connect();
+});
+
+beforeEach(async () => {
+  await DatabaseConnection.clear();
+});
+
+afterAll(async () => {
+  await DatabaseConnection.clear();
+  await DatabaseConnection.close();
+});
+
+describe('Update User Handle', () => {
+  test('Logged-in user can update their own handle', async () => {
+    const conn = await DatabaseConnection.get();
+    const member = UserFactory.fake();
+
+    await new PortalState().createUsers(member).write();
+
+    const userController = ControllerFactory.user(conn);
+    const handle = 'mycustomhandle';
+    const patchedUserResponse = await userController.patchCurrentUser({ user: { handle } }, member);
+
+    expect(patchedUserResponse.error).toBeNull();
+    const patchedUser = patchedUserResponse.user;
+    expect(patchedUser).toEqual({ ...member.getFullUserProfile(), handle });
+  });
+
+  test('User Cannot Update Their Handle to an already existing Handle', async () => {
+    const conn = await DatabaseConnection.get();
+    const [member, existingUser] = UserFactory.create(2);
+
+    await new PortalState().createUsers(member, existingUser).write();
+
+    const userController = ControllerFactory.user(conn);
+
+    const errorMessage = 'This handle is already in use.';
+    await expect(
+      userController.patchCurrentUser({ user: { handle: existingUser.handle } }, member),
+    ).rejects.toThrow(errorMessage);
+  });
+});
+
+describe('Get User Handles', () => {
+  test('User Can View Their Own Handle While Editing Profile', async () => {
+    const conn = await DatabaseConnection.get();
+    const member = UserFactory.fake();
+
+    await new PortalState().createUsers(member).write();
+
+    const userController = ControllerFactory.user(conn);
+    const userResponse = await userController.getCurrentUser(member);
+
+    expect(userResponse.error).toBeNull();
+    const { user } = userResponse;
+    expect(user.handle).toBe(member.handle);
+  });
+
+  test("Logged In User Can Fetch Another User's Profile by Handle", async () => {
+    const conn = await DatabaseConnection.get();
+    const [loggedInUser, otherUser] = UserFactory.create(2);
+
+    await new PortalState().createUsers(loggedInUser, otherUser).write();
+
+    const userController = ControllerFactory.user(conn);
+    const response = await userController.getUserByHandle(
+      { handle: otherUser.handle },
+      loggedInUser,
+    );
+
+    expect(response.user).toEqual(otherUser.getPublicProfile());
+  });
+});
+
+describe('Update to Invalid User Handle', () => {
+  test('Update handle error: too short', async () => {
+    const MIN_LENGTH = 3;
+    const user = UserFactory.fake({ handle: 'a' });
+
+    const errors = await validate(plainToClass(UserPatches, { ...user,
+      passwordChange: {
+        password: 'mypassword',
+        newPassword: 'mypassword',
+        confirmPassword: 'mypassword',
+      } }));
```

Comment:
Let's have these password changes be stored in a separate variable for improved readability, and let's add a comment on why these need to be included (i.e. mentioning that class-validator still validates optional fields for some reason and that it would error out if these fields aren't included).